packages/components/src/utils/validation.ts

```diff
@@ -0,0 +1,11 @@
+export const ALLOWED_URL_REGEXES = {
+  external: "^https:\\/\\/",
+  mail: "^mailto:",
+  internal: "^\\[.*\\]$",
+  files: ".*\\.pdf",
```

Comment:
issue: hmm I think we cannot enforce that it must be PDFs here, because we can potentially support other formats like docx, xlsx, etc.

I suggest perhaps we should consider the following format:
- External: `https://`
- Mail: `mailto:`
- Internal: `[resource:(d+):(d+)]` (I think I wrote this down in `getReferenceLinkHref`)
- Files: `/(d+)/...` (because files, like images, will be in the format `/${siteId}/${uuid}/${fileName}`)

Alternatively, since this is inside the components package, we can be less prescriptive about it and let Studio handle the pattern validation for us. The components will just assume that the link is valid and pass it through to the `href` attribute.

Reply:
quite hard to do this in studio cos we use our existing validator for the blob **and not zod**

Reply:
Sure hmm but for files need to allow other kinds of files beyond PDF